pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -225,13 +217,15 @@ def test_mutated_packages(init_and_sync):
     # Save the copy of the original packages.
     original_packages = {
         (
-            content["name"],
-            content["epoch"],
-            content["version"],
-            content["release"],
-            content["arch"],
+            content.name,
+            content.epoch,
+            content.version,
+            content.release,
+            content.arch,
         ): content
-        for content in get_content(repository.to_dict())[RPM_PACKAGE_CONTENT_NAME]
+        for content in rpm_package_api.list(
+            repository_version=repository.latest_version_href
+        ).results
```

Comment:
Q: I assume this is still paginated?  Is there an easy way to automatically de-paginate behind the scenes?

I'm pretty sure the previous helpers weren't pagination aware so it's likely not a regression / problem, just curious

Reply:
Yes, it is paginated. I am not aware of any simple way for de-paginating the results by, for instance, one parameter. @dkliban, in this case, I am required to perform the de-pagination manually with "limit=X, offset=Y" in a loop, right?

Reply:
Yeah, the API paginates with a default - 25 I think? You can do something like offset=0, limit=3million, and get everything in one page - but god help you if there are 3million packages, you'll use everyone's memory from everywhere and take forever. :)

I *think* @dralley was suggesting putting the rpm_package_api.list() call behind a generator-utility-routine, so the "deal w/ pagination" code could be "hidden" from the test-flow.

Reply:
Good. It might be interesting to write a generator fixture like this (https://github.com/pulp/pulpcore/pull/3568/files#diff-49fcc5fef555e589b9264c9c38c0e4fee041a8601cdc19bcbcd00f1fa66c9c4eR34-R47) with reasonable limit defaults:
```
    def get_upstream_distributions(self, labels=None):
        if labels:
            params = {"pulp_label_select": labels}
        else:
            params = {}
        offset = 0
        list_size = 100
        while True:
            distributions = self.distribution_ctx_cls(self.pulp_ctx).list(list_size, offset, params)
            for distro in distributions:
                yield distro
            if len(distributions) < list_size:
                break
            offset += list_size
```

Reply:
This will be tracked independently by https://github.com/pulp/pulp_rpm/issues/2981.

Reply:
Yeah to be clear that's not required for merging this PR, none of the tests would even take advantage of that at the moment, I just wanted to clarify.  But it would be nice to have